conda/fetch.py

```diff
@@ -354,19 +352,22 @@ def fetch_index(channel_urls, use_cache=False, unknown=False, index=None):
         finally:
             executor.shutdown(wait=True)
 
-    for channel, repodata in repodatas:
+    for url, repodata in repodatas:
         if repodata is None:
             continue
         new_index = repodata['packages']
-        url_s, priority = channel_urls[channel]
-        channel = channel.rstrip('/')
+        canonical_name, priority = channel_urls[url]
+        url = url.rstrip('/')
+        channel = Channel(url)
         for fn, info in iteritems(new_index):
             info['fn'] = fn
-            info['schannel'] = url_s
-            info['channel'] = channel
+            info['schannel'] = canonical_name
+            info['channel'] = url
             info['priority'] = priority
-            info['url'] = channel + '/' + fn
-            key = url_s + '::' + fn if url_s != 'defaults' else fn
+            info['url'] = url + '/' + fn
+            if channel.auth:
+                info['auth'] = channel.auth
+            key = canonical_name + '::' + fn if canonical_name != 'defaults' else fn
```

Comment:
@elehcim Feel free to review this merge commit, which is the result of merging your PR into this one. There's been a lot of refactoring to deal with other channel issues, but I think the key point in your PR is line 368 and 369 here, along with line 391 below.

For the repodata fetch, auth should most definitely now be riding along with the url.